src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node heap;
+    private Node tall;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newNode = new Node<T>(null, value, null);
+            heap = newNode;
+            tall = newNode;
+        } else {
+            Node<T> newNode = new Node<T>(tall, value, null);
+            tall.next = newNode;
+            tall = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index was failed ");
+        }
+        if (index == size) {
+            add(value);
```

Comment:
let's use `return` to finish method and then we can simplify  `else if` construction to `if`